src/main/kotlin/lotto/domain/number/Numbers.kt

```diff
@@ -0,0 +1,34 @@
+package lotto.domain.number
+
+class Numbers private constructor(val winningNumbers: List<Number>) {
+
+    companion object {
+        const val WINNING_NUMBER_COUNT = 6
+
+        fun fromInts(vararg numbers: Int): Numbers {
+            if (numbers.distinct().size != WINNING_NUMBER_COUNT) {
+                "Winning numbers must be $WINNING_NUMBER_COUNT unique values"
+            }
+
+            val numberList = numbers.map { Number(it) }
+            return Numbers(numberList)
+        }
+    }
+
+    init {
+        require(winningNumbers.size == winningNumbers.distinct().size) {
+            throw IllegalArgumentException("Winning numbers must not contain duplicates")
+        }
+        require(winningNumbers.size == WINNING_NUMBER_COUNT) {
+            throw IllegalArgumentException("Winning numbers size must equal to $WINNING_NUMBER_COUNT")
+        }
```

Comment:
```suggestion
        require(winningNumbers.size == winningNumbers.distinct().size) {
            "Winning numbers must not contain duplicates"
        }
        require(winningNumbers.size == WINNING_NUMBER_COUNT) {
            "Winning numbers size must equal to $WINNING_NUMBER_COUNT"
        }
```

It will throw `IllegalArgumentException` anyways